common/util/cmd_positional_arguments.cc

```diff
@@ -0,0 +1,128 @@
+}
+
+// Gets include directories arguments.
+std::vector<absl::string_view> CmdPositionalArguments::GetIncludeDirs() {
+  return include_dirs_;
+}
+
+std::vector<std::pair<absl::string_view, absl::string_view>>
+
+// Gets macro defines arguments.
+CmdPositionalArguments::GetDefines() {
+  return defines_;
+}
+
+// Gets SV files arguments.
+std::vector<absl::string_view> CmdPositionalArguments::GetFiles() {
+  return files_;
+}
+
+// Main function that parses arguments and add them to the correct data memeber.
+absl::Status CmdPositionalArguments::ParseArgs() {
+  // Positional arguments types:
+  // 1) <file>
+  // 2) +define+<foo>[=<value>]
+  // 3) +incdir+<dir>
+
+  int argument_index = 0;
+  for (absl::string_view argument : all_args_) {
+    if (!argument_index) {
+      argument_index++;
+      continue;
+    }  // all_args_[0] is the tool's name, which can be skipped.
+    if (argument[0] != '+') {  // the argument is a SV file name.
+      if (auto status = SetFile(argument); !status.ok())
+        return status;  // add it to the file arguments.
+    } else {            // it should be either a define or incdir.
+      std::vector<absl::string_view> argument_plus_splitted =
+          absl::StrSplit(argument, absl::ByChar('+'), absl::SkipEmpty());
+      if (argument_plus_splitted.size() < 2) {
+        // Unknown argument.
+        return absl::InvalidArgumentError("Unkown argument.");
+      }
+      absl::string_view plus_argument_type = argument_plus_splitted[0];
+      if (absl::StrContains(plus_argument_type, "define")) {
+        // define argument.
+        int define_argument_index = 0;
+        for (absl::string_view define_argument : argument_plus_splitted) {
+          if (!define_argument_index) {
+            define_argument_index++;
+            continue;
+          }
+          // define_argument is something like <macro1>=<value>.
+          std::pair<absl::string_view, absl::string_view> macro_pair =
+              absl::StrSplit(
+                  define_argument, absl::ByChar('='),
+                  absl::SkipEmpty());  // parse the macro name and value.
+          if (auto status = CmdPositionalArguments::SetDefine(macro_pair);
+              !status.ok())
+            return status;  // add the define argument.
+          define_argument_index++;
+        }
+      } else if (absl::StrContains(plus_argument_type, "incdir")) {
+        // incdir argument.
+        int incdir_argument_index = 0;
+        for (absl::string_view incdir_argument : argument_plus_splitted) {
+          if (!incdir_argument_index) {
+            incdir_argument_index++;
+            continue;
+          }
+          if (auto status =
+                  CmdPositionalArguments::SetIncludeDir(incdir_argument);
+              !status.ok())
+            return status;  // add file argument.
+          incdir_argument_index++;
+        }
+      } else {
+        return absl::InvalidArgumentError("Unkown argument.");
+      }
```

Comment:
Doesn't support macro values with `=` (e.g. `'+define+NAME="aa=bb"'`) and could be simplified a bit. Example:
```c++
#include "absl/strings/match.h"
// ...

static constexpr std::string_view kDefinePrefix = "+define+"
static constexpr std::string_view kIncDirPrefix = "+incdir+"

if (/*...*/) {
  // ...
} else if (absl::StartsWith(argument, kDefinePrefix)) {
    const auto defines = argument.substr(kDefinePrefix.size());
    std::vector<absl::string_view> defines_list = absl::StrSplit(defines, '+', absl::SkipEmpty());
    for (const auto define : defines_list) {
        // Keep '=' in macro value
        std::pair<absl::string_view, absl::string_view> macro_pair = absl::StrSplit(define, absl::MaxSplits('=', 1));
        // ...
    }

} else if (absl::StartsWith(argument, kIncDirPrefix)) {
    const auto dirname = argument.substr(kIncDirPrefix.size());
    // ...
} else {
    // ...
}
```